From 801192c00bbe09ddc8ab04e8e1a04445a7ff151f Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Wed, 18 Mar 2026 16:28:42 +0800
Subject: [PATCH 4/5] Fix aarch64 asm library install

If CMAKE_ASM_CREATE_SHARED_LIBRARY is undefined, then no shared
libs will be built if all of the sources are assembly files.
---
 folly/external/aor/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/folly/external/aor/CMakeLists.txt b/folly/external/aor/CMakeLists.txt
index c7b6e9338..c2cdc8946 100644
--- a/folly/external/aor/CMakeLists.txt
+++ b/folly/external/aor/CMakeLists.txt
@@ -70,4 +70,8 @@ folly_add_library(
     -DFOLLY_MEMSET_IS_MEMSET
 )
 
+if(BUILD_SHARED_LIBS)
+  set(CMAKE_ASM_CREATE_SHARED_LIBRARY ${CMAKE_C_CREATE_SHARED_LIBRARY})
+endif()
+
 endif()
-- 
2.52.0

